app/src/main/java/com/example/byitscover/scrapers/GoodreadsScraper.java

```diff
@@ -108,10 +109,32 @@ public GoodreadsScraper() {
                 rating,
                 null,
                 reviews,
-                coverUrl);
+                coverUrl,
+                getPrice(bookDocument));
 
         List<BookListing> listings = new ArrayList<BookListing>();
         listings.add(listing);
         return listings;
     }
+
+    /**
+     * This method returns the price found on the website
+     * @return price
+     */
+    private BigDecimal getPrice(Document bookDocument) {
+        String priceString;
+        try {
+            Element priceValue = bookDocument.getElementById("buyButtonContainer");
+            priceString = priceValue.childNode(3).childNode(0).childNode(0).childNode(0).toString();
+            priceString = priceString.substring(priceString.lastIndexOf('$') + 1)
+                    .replaceAll("\\s+", "");
+            if (!priceString.contains(".")) {
+                priceString = priceString.substring(0, priceString.length() - 2)
+                        + "." + priceString.substring(priceString.length() - 2);
+            }
+        } catch (Exception ex) {
+            priceString = "0.00";
```

Comment:
Maybe we should represent unknown prices with null. Some websites have free e-book listings.

Reply:
Updated